### DISCUSSION TITLE
Breakdown by multiple properties with one Stats API call

### DISCUSSION THREAD

**Discussion by RobertJoonas** [plausible/analytics#1254] [Anything Else]
It's currently not possible to get a breakdown of multiple properties with one API call.

For example, querying with two properties `event:page` and `visit:source` could give the following breakdown:

page: /; source: Google; visitors: 123
page: /; source: Twitter; visitors: 113
page: /info; source: Google; visitors: 103
page: /info; source: Twitter; visitors: 103
etc...

Currently this can only be simulated by making separate API calls, but it would be great if users could get this by only using a single query.

(feedback via email)

**Comment by RobertJoonas** [comment]
One more vote on this via email - breakdown by a time period + property

"To analyse and optimize campaigns we would need a breakdown by day AND utm_content for visits and conversions."

**Comment by giocomai** [comment]
Do I understand it correctly that if I need a table with the number of visitors from each source to each page for a given set of dates I need to make a bunch of separate queries, more or less as follows:

1. First query API for `visit:source` filtering on a single day. 
2. Iterate on each source, filtering for that source and that day, keeping the breakdown for `event:page`
3. Combine the result
4. Repeat for each day

This means that, for example, if I have on average incoming traffic from 50 different sources each day, then I will have to make:
1 api call (step 1) + 50 api calls (step 2)  = 51 api calls per day, which then needs to be multiplied for the number of days. 

I've implemented this and am caching the results locally, so once it's setup, in my use case, it's still manageable. But after I set up the whole thing I looked at it and thought that perhaps I am missing something and there is a better way to go about it. 

Or based on the current API, is this really more or less the way to go about it?

Using the timeseries end point and filtering by each combination of page and source may or may not be more efficient, depending on the length of the time series.
